confmap/provider/secretsmanagerprovider/provider.go

```diff
@@ -40,29 +47,56 @@ func newWithSettings(_ confmap.ProviderSettings) confmap.Provider {
 //
 // Deprecated: [v0.100.0] Use NewFactory() instead.
 func New() confmap.Provider {
-	return &provider{}
+	return &provider{client: nil}
 }
 
 func (provider *provider) Retrieve(ctx context.Context, uri string, _ confmap.WatcherFunc) (*confmap.Retrieved, error) {
 	if !strings.HasPrefix(uri, schemeName+":") {
 		return nil, fmt.Errorf("%q uri is not supported by %q provider", uri, schemeName)
 	}
 
-	secretArn := strings.Replace(uri, schemeName+":", "", 1)
+	// initialize the secrets manager client in the first call of Retrieve
```

Comment:
that's not threadsafe, is there a way to do that in the Start method?

Reply:
It can be moved to `New()` and `newWithSettings()` but I see similar pattern in other confmap providers e.g. [s3provider](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/confmap/provider/s3provider/provider.go#L79), [httpsprovider](https://github.com/open-telemetry/opentelemetry-collector/blob/main/confmap/provider/internal/configurablehttpprovider/provider.go#L88).

Reply:
no, moving to initialization might not work either. Maybe we need to introduce more lifecycle hooks to providers.